pulumi/infra/nomad_job.py

```diff
@@ -28,11 +28,13 @@ def __init__(
             resource_name=f"{DEPLOYMENT_NAME}-{name}-job",
             jobspec=self._file_contents(str(jobspec)),
             hcl2=nomad.JobHcl2Args(enabled=True, vars=self._fix_pulumi_preview(vars)),
-            opts=pulumi.ResourceOptions(parent=self),
             # Wait for all services to become healthy
             detach=False,
             # Purge job from Nomad servers after a `pulumi destroy`
             purge_on_destroy=True,
+            opts=pulumi.ResourceOptions(
+                parent=self, custom_timeouts=opts.custom_timeouts if opts else None
```

Comment:
What may be better than is something like
`pulumi.ResourceOptions.merge(opts, pulumi.ResourceOptions(parent=self))` that way if anything else is passed to opts it gets inherited by this as well.

Reply:
i like it